CFC_DataCollector/trip_analysis/similarity.py

```diff
@@ -0,0 +1,169 @@
+                start_lat = self.data[b]['trip_start_location'][1]
+                start_lon = self.data[b]['trip_start_location'][0]
+                end_lat = self.data[b]['trip_end_location'][1]
+                end_lon = self.data[b]['trip_end_location'][0]
+                path = [(start_lat, start_lon), (end_lat, end_lon)]
+                mymap.addpath(path, matcol.rgb2hex(colormap(float(self.bins.index(bin))/len(self.bins))))
+        mymap.draw('./mybins.html')
+        
+    #evaluate the bins as if they were a clustering on the data
+    def evaluate_bins(self):
+        self.labels = []
+        newcolors = []
+        for bin in self.bins:
+            for b in bin:
+                self.labels.append(self.bins.index(bin))
+                if self.colors != None:                
+                    newcolors.append(self.colors[b])
+        self.colors = newcolors
+                    
+        labels = numpy.array(self.labels)
+        colors = numpy.array(self.colors)
+        points = []
+        for bin in self.bins:
+            for b in bin:
+                start_lat = self.data[b]['trip_start_location'][1]
+                start_lon = self.data[b]['trip_start_location'][0]
+                end_lat = self.data[b]['trip_end_location'][1]
+                end_lon = self.data[b]['trip_end_location'][0]
+                path = [start_lat, start_lon, end_lat, end_lon]
+                points.append(path)
+
+        if self.colors != None:
+            a = metrics.silhouette_score(numpy.array(points), labels)
+            b = homogeneity_score(colors, labels)
+            c = completeness_score(colors, labels)
+            
+            print 'number of bins is ' + str(len(self.bins))
+            print 'silhouette score is ' + str(a)
+            print 'homogeneity is ' + str(b)
+            print 'completeness is ' + str(c)
+            print 'accuracy is ' + str(((a+1)/2.0 + b + c)/3.0)
+
+    #update the ground truth
+    def update_colors(self, newcolors):
+        if self.colors != None:
+            indices = [] * len(set(newcolors))
+            for n in newcolors:
+                if n not in indices:
+                    indices.append(n)
+            for i in range(len(newcolors)):
+                newcolors[i] = indices.index(newcolors[i])
+            self.colors = newcolors
+
+
+    #calculate the distance between two trips
+    def distance_helper(self, a, b):
+        starta = self.data[a]['trip_start_location']
+        startb = self.data[b]['trip_start_location']
+        enda = self.data[a]['trip_end_location']
+        endb = self.data[b]['trip_end_location']
+
+        start = self.distance(starta[1], starta[0], startb[1], startb[0])
+        end = self.distance(enda[1], enda[0], endb[1], endb[0])
+        if start and end:
+            return True
+        return False
+
+    #calculate the meter distance between two trips
+    def distance(self, lat1, lon1, lat2, lon2, option=False):
```

Comment:
Again, I think that there is an existing function that calculates the distance between two points.